Content.Client/UserInterface/Systems/Storage/Controls/ItemGridPiece.cs

```diff
@@ -160,14 +160,16 @@ protected override void Draw(DrawingHandleScreen handle)
         }
 
         // typically you'd divide by two, but since the textures are half a tile, this is done implicitly
-        var iconPosition = new Vector2((boundingGrid.Width + 1) * size.X + itemComponent.StoredOffset.X * 2,
-            (boundingGrid.Height + 1) * size.Y + itemComponent.StoredOffset.Y * 2);
+        var iconOffset = Location.Rotation.RotateVec(itemComponent.StoredOffset);
+        var iconPosition = new Vector2(
+            (boundingGrid.Width + 1) * size.X + iconOffset.X * 2,
+            (boundingGrid.Height + 1) * size.Y + iconOffset.Y * 2);
         var iconRotation = Location.Rotation + Angle.FromDegrees(itemComponent.StoredRotation);
 
         if (itemComponent.StoredSprite is { } storageSprite)
         {
             var scale = 2 * UIScale;
-            var offset = (((Box2) boundingGrid).Size - Vector2.One) * size;
+            var offset = (((Box2) boundingGrid).Size - Vector2.One) * size + iconOffset;
```

Comment:
something is wrong here with the offset
I edited both entities using VV to set their `StoredOffset` to 20
![test](https://github.com/user-attachments/assets/d1a4df56-b653-4fed-90ad-545eb4403b72)
is should not get multiplied by the scale here I think

Reply:
huh